### ISSUE TITLE
clientFromString unix "path" argument undocumented and not positional

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5640]
|<img alt="magmatt's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| magmatt reported|
|-|-|
|Trac ID|trac#5640|
|Type|enhancement|
|Created|2012-04-12 15:09:48Z|

When creating a UNIX socket client using endpoints.clientFromString, the "path" argument is not documented and is required to be a keyword (instead of positional, like for TCP's host)

Here's the problem:

```
>>> from twisted.internet.endpoints import clientFromString
>>> clientFromString(object(), 'unix:/var/foo/bar')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/site-packages/Twisted-10.2.0-py2.6-macosx-10.3-fat.egg/twisted/internet/endpoints.py", line 1005, in clientFromString
    kwargs = _clientParsers[name](*args, **kwargs)
TypeError: _parseClientUNIX() takes exactly 0 arguments (1 given)
>>> clientFromString(object(), 'unix:host=/var/foo/bar')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/2.6/lib/python2.6/site-packages/Twisted-10.2.0-py2.6-macosx-10.3-fat.egg/twisted/internet/endpoints.py", line 1006, in clientFromString
    return _endpointClientFactories[name](reactor, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'host'
>>> clientFromString(object(), 'unix:path=/var/foo/bar')
<twisted.internet.endpoints.UNIXClientEndpoint object at 0x3a91b0>
```

I will attach a patch that adds documentation and allows the "path" argument to be positionally given.

Attachments:

* [unix-client-patch.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/351/3511af4942df57122427095bcebc72ccf1f20cbd/28e80b5746d12481ebd95c973318b4b179ee4ed7.patch) (3048 bytes) - added by magmatt on 2012-04-12 15:10:43Z - Patch for docs and positional path arg

<details><summary>Searchable metadata</summary>

```
trac-id__5640 5640
type__enhancement enhancement
reporter__magmatt magmatt
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1334243388000000 1334243388000000
changetime__1334670596000000 1334670596000000
version__None None
owner__exarkun exarkun
cc__magmatt
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

Thanks!  This patch looks really good.  The only tiny issue I see is that the new test method makes the same mistake as many of the existing test methods: in its document, it says _L{endpointClient}_, which I think was an old in-development-only name of `clientFromString`.  I'll adjust the docstring of all the test methods that make this mistake, add a news file, and apply the change.  Thanks again.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [34222]) Add support for positional path values to the UNIX client endpoint string description syntax

Author: magmatt
Reviewer: exarkun
Fixes: [#5640](https://github.com/twisted/twisted/issues/5640)
